test/e2e/ingress.go

```diff
@@ -243,31 +374,68 @@ var _ = Describe("GCE L7 LoadBalancer Controller", func() {
 		framework.beforeEach()
 		client = framework.Client
 		ns = framework.Namespace.Name
-		Expect(waitForRCPodsRunning(client, "kube-system", "glbc")).NotTo(HaveOccurred())
-		Expect(checkLeakedResources()).NotTo(HaveOccurred())
+		// Scaled down the existing Ingress controller so it doesn't interfere with the test.
+		Expect(scaleKubeSystemRC(client, lbContainerName, 0)).NotTo(HaveOccurred())
```

Comment:
Sure, my concern is that we could be running e2e tests at a different version than the cluster is running. We do this, e.g. when testing upgrades. If we launch a different version of the lbcontainer we aren't testing the actual cluster functionality unless we verify that the pod we launch is actually the same as the kube-system one. But we can't verify that if the lbcontainer version changes across a cluster upgrade.